languages/python/django-oso/tests/test_post_relationship.py

```diff
@@ -579,18 +579,138 @@ def test_reverse_many_relationship(tag_nested_many_many_fixtures):
+                      FROM "test_app2_post" W0
+                      INNER JOIN "test_app2_user" W1 ON (W0."created_by_id" = W1."id")
+                      LEFT OUTER JOIN "test_app2_user_posts" W2 ON (W1."id" = W2."user_id")
+                      WHERE (EXISTS(SELECT V0."id"
+                                    FROM "test_app2_post" V0
+                                    INNER JOIN "test_app2_user" V1 ON (V0."created_by_id" = V1."id")
+                                    LEFT OUTER JOIN "test_app2_user_posts" V2 ON (V1."id" = V2."user_id")
+                                    WHERE (EXISTS(SELECT U0."id"
+                                                  FROM "test_app2_post" U0
+                                                  INNER JOIN "test_app2_user" U1 ON (U0."created_by_id" = U1."id")
+                                                  INNER JOIN "test_app2_user_posts" U2 ON (U1."id" = U2."user_id")
+                                                  WHERE (U0."id" = V2."post_id"
+                                                         AND U0."id" > 3
+                                                         AND U0."id" = U2."post_id"))
+                                           AND V0."id" = W2."post_id"
+                                           AND V0."id" > 2))
+                             AND W0."id" = "test_app2_user_posts"."post_id"
+                             AND W0."id" > 1))
+               AND "test_app2_post"."id" > 4)
+    """
+    assert str(posts.query) == " ".join(expected.split())
+    assert len(posts) == 1
+
+@pytest.mark.django_db
+def test_unify_ins(tag_nested_many_many_fixtures):
+    Oso.load_str(
+        """
+        allow(_, _, post) if
+            user1 in post.users and
+            user2 in post.users and
+            user1.id = user2.id and
+            user1.id > 1 and
+            user2.id <= 2;
+        """
+    )
+    user = User.objects.get(username="user")
+    authorize_filter = authorize_model(None, Post, actor=user, action="read")
+    posts = Post.objects.filter(authorize_filter)
+    expected = """
+        SELECT "test_app2_post"."id", "test_app2_post"."contents", "test_app2_post"."access_level", "test_app2_post"."created_by_id", "test_app2_post"."needs_moderation"
+        FROM "test_app2_post"
+        LEFT OUTER JOIN "test_app2_user_posts" ON ("test_app2_post"."id" = "test_app2_user_posts"."post_id")
+        WHERE (EXISTS(SELECT U0."id"
+                      FROM "test_app2_user" U0
+                      WHERE (U0."id" = "test_app2_user_posts"."user_id"
+                             AND U0."id" = "test_app2_user_posts"."user_id"
+                             AND U0."id" = "test_app2_user_posts"."user_id"
+                             AND U0."id" > 1
+                             AND U0."id" <= 2))
+               AND EXISTS(SELECT U0."id"
+                          FROM "test_app2_user" U0
+                          WHERE (U0."id" = "test_app2_user_posts"."user_id"
```

Comment:
We are generating 3 identical clauses in this where.

Reply:
We _are_ asking it to do this three times, so kinda makes sense.

Reply:
updated to not do this